### ISSUE TITLE
How to ensure all search() history messages are unique

### ISSUE THREAD

**Issue by logan-jobzmall** [pubnub/chat-engine#274]
@ianjennings,

I am in the middle of getting infinite scroll working with the search() method (being able to scroll up through all the history). How do I ensure that the messages that are coming in are unique? It seems as though search() is coming back with duplicate messages unconditionally once it gets to the last message. 

```
room.search = () => {
                room.loadingMessages = true;
                room.chat.search({event: 'message'})
                    .plugin(ChatEngineCore.plugin['chat-engine-markdown']())
                    .plugin(ChatEngineCore.plugin['chat-engine-unread-messages']())
                    .plugin(ChatEngineCore.plugin['chat-engine-emoji']({
                        url: 'https://www.webpagefx.com/tools/emoji-cheat-sheet/graphics/emojis'
                    }))
                    .on('message', function (payload) {
                        addMessage(payload, 'history');
                    }).on('upload', function (payload) {
                        addMessage(payload, 'upload');
                    })
                    .on('$.search.finish', () => {
                        room.loadingMessages = false;
                    })
            }
```
Logan

**Comment by deepak1725** [comment]
@logan-jobzmall .. how about mutating after fire function and adding `Id` param in payload in Pubnub Admin panel..??

**Comment by ianjennings** [comment]
So even after "search.finish", you continue to get messages?

**Comment by logan-jobzmall** [comment]
@deepak1725 - That sounds like an idea! I will need to look into that.

@ianjennings - Once search hits it's "limit" it just continues to re-pull from the beginning again when search() is called. Don't know if that was intended or if it was supposed to not return anything after that again from the backend.

**Comment by ianjennings** [comment]
Hey @logan-jobzmall, a fix for this has been released on ChatEngine version ```0.9.5```! It's on NPM. Thanks for reporting this issue.
